libstb/secvar/storage/secboot_tpm.c

```diff
@@ -421,94 +494,75 @@ static int secboot_tpm_store_init(void)
-		rc = secboot_format();
-		if (rc)
-			goto error;
-
-		/* Everything just got reformatted, so we're done here */
+		/* Indicies got defined and formatted, we're done here */
 		goto done;
 	} else if (vars_defined ^ control_defined) {
 		/* This should never happen. Both indices should be defined at the same
-		 * time. Otherwise something seriously went wrong. P A N I C. */
-		// TODO: panic
-	}
-
-	/* Ensure the NV indices were defined with the correct set of attributes, hierarchy */
-	/* TODO: Do we want to undefine here, or panic and have them assert physical presence? */
-	rc = tss_nv_read_public(SECBOOT_TPMNV_VARS_INDEX, &nv_public, &nv_name);
-	if (rc)
-		goto error;
-
-	if (memcmp(tpmnv_vars_name, &nv_name, sizeof(tpmnv_vars_name))) {
+		 * time. Otherwise something seriously went wrong. */
+		prlog(PR_ERR, "NV indices defined with unexpected attributes. Assert physical presence to clear\n");
 		goto error;
 	}
 
-	rc = tss_nv_read_public(SECBOOT_TPMNV_CONTROL_INDEX, &nv_public, &nv_name);
-	if (rc)
-		goto error;
-
-	if (memcmp(tpmnv_control_name, &nv_name, sizeof(tpmnv_control_name))) {
-		goto error;
-	}
+	/* Ensure the NV indices were defined with the correct set of attributes */
+	secboot_tpm_check_tpmnv_attrs();
 
-	/* TPMNV indices exist and weren't just formatted, so read them in */
+	/* TPMNV indices exist, are correct, and weren't just formatted, so read them in */
 	rc = tpmnv_ops.read(SECBOOT_TPMNV_VARS_INDEX, tpmnv_vars_image, tpmnv_vars_size, 0);
-	if (rc)
+	if (rc) {
+		prlog(PR_ERR, "Failed to read from the VARS index\n");
 		goto error;
+	}
 
 	rc = tpmnv_ops.read(SECBOOT_TPMNV_CONTROL_INDEX, tpmnv_control_image, sizeof(struct tpmnv_control), 0);
-	if (rc)
+	if (rc) {
+		prlog(PR_ERR, "Failed to read from the CONTROL index\n");
 		goto error;
+	}
 
+	/* Verify the header information is correct */
 	if (tpmnv_vars_image->header.magic_number != SECBOOT_MAGIC_NUMBER ||
-	    tpmnv_control_image->header.magic_number != SECBOOT_MAGIC_NUMBER) {
-		prlog(PR_ERR, "CRITICAL: TPMNV indices defined, but contain bad data. Assert physical presence to clear\n");
+	    tpmnv_control_image->header.magic_number != SECBOOT_MAGIC_NUMBER ||
+	    tpmnv_vars_image->header.version != SECBOOT_VERSION ||
+	    tpmnv_control_image->header.version != SECBOOT_VERSION) {
+		prlog(PR_ERR, "TPMNV indices defined, but contain bad data. Assert physical presence to clear\n");
 		goto error;
 	}
 
-	/* Determine if we need to reformat the secboot PNOR partition */
-	if (secboot_image->header.magic_number != SECBOOT_MAGIC_NUMBER) {
-		rc = secboot_format();
-		if (rc)
-			goto error;
+	/* Verify the secboot partition header information, reformat if incorrect
+	 * Note: Future variants should attempt to handle older versions safely
+	 */
+	if (secboot_image->header.magic_number != SECBOOT_MAGIC_NUMBER ||
+	    secboot_image->header.version != SECBOOT_VERSION) {
+		secboot_format();
```

Comment:
Again return code here ?

Reply:
good catch, i removed those checks when i made them panic. forgot to re-add the checks